src/category_theory/over.lean

```diff
@@ -130,6 +131,41 @@ instance forget_reflects_iso : reflects_isomorphisms (forget X) :=
 { reflects := λ Y Z f t, by exactI
   { inv := over.hom_mk t.inv ((as_iso ((forget X).map f)).inv_comp_eq.2 (over.w f).symm) } }
 
+instance forget_faithful : faithful (forget X) := {}.
+
+/--
+If `k.left` is an epimorphism, then `k` is an epimorphism. In other words, `over.forget X` reflects
+epimorphisms.
+The converse does not hold without additional assumptions on the underlying category.
+-/
+-- TODO: Show the converse holds if `T` has binary products or pushouts.
+lemma epi_of_epi_left {f g : over X} (k : f ⟶ g) [hk : epi k.left] : epi k :=
+faithful_reflects_epi (forget X) hk
```

Comment:
By the way @rwbarton I saw `faithful_reflects_epi` and related take `epi` as a `()` argument rather than `[]`, and that I think you mentioned in the past that that's a better idea. Do you think we should make this change throughout category theory in mathlib? 
For the record I'm neutral on this, but the inconsistency is getting frustrating.

Reply:
I'm also neutral on this. If we switch to `()` it would be good to refactor the names as well, to take advantage of dot-notation.